Remotion/Web/Core/ContentSecurityPolicy/CspEnabledHtmlTextWriter.cs

```diff
@@ -29,18 +29,94 @@ namespace Remotion.Web.ContentSecurityPolicy
   /// </summary>
   public class CspEnabledHtmlTextWriter : HtmlTextWriter
   {
-    private static readonly IReadOnlyDictionary<string, string> s_supportedEvents =
-        new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+    /// <summary>
+    /// Registers the specified <paramref name="eventName"/> as a supported event.
+    /// The default list of events should be complete, but this method can be used in case it is not.
+    /// </summary>
+    /// <param name="eventName">The name of the attribute event (e.g. onclick).</param>
+    public static void RegisterSupportedEvent (string eventName)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty(nameof(eventName), eventName);
+      if (!eventName.StartsWith("on", StringComparison.OrdinalIgnoreCase))
+        throw new ArgumentException("The specified event name must start with 'on'.", nameof(eventName));
+
+      s_supportedEvents.TryAdd(eventName, eventName);
+    }
+
+    private static readonly Dictionary<string, string> s_supportedEvents =
```

Comment:
In order to support a static Register we need to make the Dictionary into a ConcurrectDict